### PR TITLE
UI Integration  to add multiple tags, terms and owners

### PR DESCRIPTION
**Pull request by Ankit-Keshari-Vituity**
Adding UI Integration for adding multiple tags, terms, and owners

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+            if (e instanceof Error) {
+                message.error({ content: `Failed to add owners: \n ${e.message || ''}`, duration: 3 });
+            }
+        } finally {
+            refetch?.();
+            onModalClose();
+        }
+    };
+
+    console.log('combinedSearchResults', combinedSearchResults);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx:221]: Remove!

File: datahub-web-react/src/app/shared/SelectedOwnerTag.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import { Tag } from 'antd';
+
+type Props = {
+    closable: boolean;
+    onClose: (event?: React.MouseEvent<HTMLElement, MouseEvent> | undefined) => void;
+    label: React.ReactNode;
+};
+
+export default function SelectedOwnerTag({ closable, onClose, label }: Props) {
+    console.log('label:: ', label);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/SelectedOwnerTag.tsx:11]: remove

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+
+const getSelectedValue = (rawValue: string) => {
+    const [name, type] = rawValue.split(NAME_TYPE_SEPARATOR);
+    return {
+        name,
+        type,
+    };
+};
+
+const renderTerm = (suggestion: string, icon: JSX.Element, type: string) => ({
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx:57]: Are there existing components we can use?

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+    label: (
+        <SuggestionContainer>
+            <BookOutlined style={{ marginRight: '3%' }} />
+            <SuggestionText>{suggestion}</SuggestionText>
+        </SuggestionContainer>
+    ),
+    type,
+});
+
+const renderTag = (suggestion: string, $colorHash, $color, type: string) => ({
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx:68]: Are there any existing components we can use? If not we should create one!

File: datahub-web-react/src/app/shared/SelectedOwnerTag.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { Tag } from 'antd';
+
+type Props = {
+    closable: boolean;
+    onClose: (event?: React.MouseEvent<HTMLElement, MouseEvent> | undefined) => void;
+    label: React.ReactNode;
+};
+
+export default function SelectedOwnerTag({ closable, onClose, label }: Props) {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/SelectedOwnerTag.tsx:10]: OwnerPill

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx (hunk 1 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx]: Removed and committed!

File: datahub-web-react/src/app/shared/SelectedOwnerTag.tsx (hunk 2 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/shared/SelectedOwnerTag.tsx]: Removed and committed

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx (hunk 3 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx:41]: Created a new component TermPill and returned the JSX as per requirement.

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx (hunk 4 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx:47]: Created a new component TagPill and returned the JSX as per requirement.

File: datahub-web-react/src/app/shared/SelectedOwnerTag.tsx (hunk 5 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/shared/SelectedOwnerTag.tsx]: Created a new component OwnerPill and returned the JSX as per requirement.

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx (hunk 3 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx:41]: committed

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx (hunk 4 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx:47]: committed

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+    type: EntityType;
+    visible: boolean;
+    defaultOwnerType?: OwnershipType;
+    hideOwnerType?: boolean | undefined;
+    onCloseModal: () => void;
+    refetch?: () => Promise<any>;
+};
+
+type SelectedOwner = {
+    label: string;
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx:53]: It is not clear from this type what the expected values of "label" and "value" are -- is there any way to improve the names of this type?

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+            .filter((result) => result.entity.urn === newSelectedOwner.value)
+            .map((result) => result.entity);
+        if (filteredActors.length) {
+            const actor = filteredActors[0];
+            const ownerEntityType =
+                actor && actor.type === EntityType.CorpGroup ? OwnerEntityType.CorpGroup : OwnerEntityType.CorpUser;
+            const newValues = [
+                ...selectedOwners,
+                {
+                    label: newSelectedOwner.value,
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx:100]: Let's add a comment to say that the "value" field will be the urn of the owner.

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+    visible: boolean;
+    defaultOwnerType?: OwnershipType;
+    hideOwnerType?: boolean | undefined;
+    onCloseModal: () => void;
+    refetch?: () => Promise<any>;
+};
+
+type SelectedOwner = {
+    label: string;
+    value: any;
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx:54]: Can we make this type not "any" but actually the following: 

{
  ownerUrn: string ,
  ownerEntityType: EntityType,

}

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+                            value={selectedOwners}
+                            autoFocus
+                            mode="multiple"
+                            filterOption={false}
+                            placeholder="Search for users or groups..."
+                            onSelect={(asset: any) => onSelectOwner(asset)}
+                            onDeselect={(asset: any) => onDeselectOwner(asset)}
+                            onSearch={handleActorSearch}
+                            tagRender={(tagProps) => (
+                                <OwnerPill
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx:253]: Nice!

File: datahub-web-react/src/app/policy/PolicyActorForm.tsx [hunk 10]
```diff
@@ -144,15 +153,29 @@ export default function PolicyActorForm({ policyType, actors, setActors }: Props
...
             <SearchResultContainer>
-                <ActorName>{entityRegistry.getDisplayName(result.entity.type, result.entity)}</ActorName>
                 <Link
                     target="_blank"
                     rel="noopener noreferrer"
                     to={() => `/${entityRegistry.getPathName(result.entity.type)}/${result.entity.urn}`}
                 >
-                    View
+                    <SearchResultContent>
+                        <CustomAvatar
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/policy/PolicyActorForm.tsx:169]: Can we reuse ownerPill here?

File: datahub-web-react/src/app/policy/PolicyActorForm.tsx [hunk 11]
```diff
@@ -199,9 +222,7 @@ export default function PolicyActorForm({ policyType, actors, setActors }: Props
                     onDeselect={(asset: any) => onDeselectUserActor(asset)}
                     onSearch={handleUserSearch}
                     tagRender={(tagProps) => (
-                        <Tag closable={tagProps.closable} onClose={tagProps.onClose}>
-                            {tagProps.value}
-                        </Tag>
+                        <OwnerPill closable={tagProps.closable} onClose={tagProps.onClose} label={tagProps.label} />
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/policy/PolicyActorForm.tsx:225]: Nice!!!

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx (hunk 3 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx:41]: amazing!

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx [hunk 12]
```diff
@@ -0,0 +1,327 @@
...
+                onClose={onCloseModal}
+                onBack={() => setShowCreateModal(false)}
+                tagName={inputValue}
+                entityUrn={entityUrn}
+                entitySubresource={entitySubresource}
+            />
+        );
+    }
+
+    const getUrnId = (urnId: string) => {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx:179]: Let's rename all instances of "urnId" to just "urn"

An urn is implied to be an ID.

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx [hunk 13]
```diff
@@ -0,0 +1,327 @@
...
+        const { name: selectedName, type: selectedType } = getSelectedValue(urnId);
+        let tempUrn = '';
+        if (selectedType === EntityType.Tag) {
+            tempUrn = `urn:li:tag:${selectedName}`;
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx:183]: I'm guessing we already do this in existing code?

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx [hunk 14]
```diff
@@ -0,0 +1,327 @@
...
+        return tempUrn;
+    };
+
+    // When a Tag or term search result is selected, add the urn to the UrnIds
+    const onSelectValue = (newUrnId: string) => {
+        if (newUrnId === CREATE_TAG_VALUE) {
+            setShowCreateModal(true);
+            return;
+        }
+        const newUrnIds = [...(urnIds || []), getUrnId(newUrnId)];
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx:197]: Let's rename UrnIds to just Urns, if this array just holds URNs.

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx [hunk 15]
```diff
@@ -0,0 +1,327 @@
...
+            onCloseModal();
+            return;
+        }
+        setDisableAdd(true);
+
+        let input = {};
+        let actionType = EntityActionType.UpdateSchemaTags;
+        if (type === EntityType.Tag) {
+            input = {
+                tagUrns: urnIds,
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx:227]: Again, let's never use the name "urnIds" --> URNs are already implying that they are ids.

Just call them "urns"

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx (hunk 9 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx:253]: Thank you!

File: datahub-web-react/src/app/policy/PolicyActorForm.tsx (hunk 11 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/policy/PolicyActorForm.tsx:225]: Thank you!

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx (hunk 14 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx]: Changed and committed

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx (hunk 15 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx]: Changed and committed

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx (hunk 7 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx:100]: Added and committed

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx (hunk 14 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx]: Changed and Committed my changes

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx (hunk 13 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx:205]: No, this is a new helper function to get the urn from the selected value.

File: datahub-web-react/src/app/policy/PolicyActorForm.tsx (hunk 11 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/policy/PolicyActorForm.tsx:225]: Thank you !

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx (hunk 9 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx:253]: Thank you !

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx (hunk 8 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx]: The value accepting in select input field is in {label: string, value: string} format

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx (hunk 7 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx]: Added and Committed

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx (hunk 6 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx:52]: Added and Committed

**Review by jjoyce0510 — APPROVED** [review]

**Pull request merged**